showkase/src/main/java/com/airbnb/android/showkase/ui/ShowkaseCategoriesScreen.kt

```diff
@@ -34,60 +37,62 @@ internal fun ShowkaseCategoriesScreen(
                 SimpleTextCard(
                     text = "$title ($categorySize)",
                     onClick = {
-                        showkaseBrowserScreenMetadata.update {
-                            copy(
+                        onUpdateShowkaseBrowserScreenMetadata(
+                            showkaseBrowserScreenMetadata.copy(
                                 currentGroup = null,
                                 isSearchActive = false,
                                 searchQuery = null
                             )
-                        }
+                        )
                         when (category) {
-                            ShowkaseCategory.COMPONENTS -> navController.navigate(
-                                ShowkaseCurrentScreen.COMPONENT_GROUPS
-                            )
-                            ShowkaseCategory.COLORS -> navController.navigate(
-                                ShowkaseCurrentScreen.COLOR_GROUPS
-                            )
-                            ShowkaseCategory.TYPOGRAPHY -> navController.navigate(
-                                ShowkaseCurrentScreen.TYPOGRAPHY_GROUPS
-                            )
+                            ShowkaseCategory.COMPONENTS -> onNavigateToComponentGroups()
```

Comment:
It would be nice if we could check here whether the component group only has one component and directly navigate into it.